modules/sovrnBidAdapter.js

```diff
@@ -22,48 +24,58 @@ export const spec = {
    * @return object of parameters for Prebid AJAX request
    */
   buildRequests: function(bidReqs, bidderRequest) {
-    const loc = utils.getTopWindowLocation();
-    let sovrnImps = [];
-    let iv;
-    utils._each(bidReqs, function (bid) {
-      iv = iv || utils.getBidIdParameter('iv', bid.params);
-      sovrnImps.push({
-        id: bid.bidId,
-        banner: { w: 1, h: 1 },
-        tagid: String(utils.getBidIdParameter('tagid', bid.params)),
-        bidfloor: utils.getBidIdParameter('bidfloor', bid.params)
+    try {
+      const loc = utils.getTopWindowLocation();
+      let sovrnImps = [];
+      let iv;
+      utils._each(bidReqs, function (bid) {
+        iv = iv || utils.getBidIdParameter('iv', bid.params);
+        bid.sizes = ((utils.isArray(bid.sizes) && utils.isArray(bid.sizes[0])) ? bid.sizes : [bid.sizes])
+        bid.sizes = bid.sizes.filter(size => utils.isArray(size))
```

Comment:
do we need this filter? what are all the forms sizes can take? it's really too bad pre bid doesn't standardize this structure

Reply:
The filter protects against crashing if the user provides an invalid or null value for bid.sizes.

Might as well be careful.